drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionViewImpl.java

```diff
@@ -148,12 +192,53 @@ public void initListStructure(String key, Map<String, String> simplePropertiesMa
      * @param key The key representing the property, i.e Classname#propertyname (e.g Author#books)
      * @param keyPropertyMap
      * @param valuePropertyMap
+     * @param isRule
+     * @param isExpression
+     *
      */
     @Override
-    public void initMapStructure(String key, Map<String, String> keyPropertyMap, Map<String, String> valuePropertyMap) {
+    public void initMapStructure(String key, Map<String, String> keyPropertyMap, Map<String, String> valuePropertyMap, boolean isRule, boolean isExpression) {
+        listWidget = false;
+        commonInit(isRule, isExpression);
+        createLabel.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.createLabelMap());
+        collectionCreationModeLabel.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.collectionMapCreation());
+        collectionCreationCreateLabel.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.createLabelMap());
+        collectionCreationCreateSpan.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.createLabelMapDescription());
+        collectionCreationDefineLabel.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.defineLabelMap());
+        collectionCreationDefineSpan.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.defineLabelMapDescription());
         presenter.initMapStructure(key, keyPropertyMap, valuePropertyMap, this);
     }
 
+    protected void commonInit(boolean isRule, boolean isExpression) {
+        ruleScenario = isRule;
+        saveButton.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.saveButton());
+        cancelButton.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.cancelButton());
+        removeButton.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.removeButton());
+        addItemButtonLabel.setInnerText(ScenarioSimulationEditorConstants.INSTANCE.collectionEditorAddNewItem());
+        if (isExpression) {
+            defineCollectionRadio.setChecked(true);
+            enableCreateCollectionContainer(false);
+        } else {
+            createCollectionRadio.setChecked(true);
+            enableCreateCollectionContainer(true);
+        }
+        if (isRule) {
+            initAndRegisterHandlerForExpressionTextArea();
+        }
+    }
+
+    /**
+     * It inits and registers the native "input" , which is not managed by GWT
+     */
+    protected void initAndRegisterHandlerForExpressionTextArea() {
+        checkExpressionSyntax();
+        DOM.sinkBitlessEvent(expressionElement, ConstantHolder.INPUT);
+        DOM.setEventListener(expressionElement, event -> {
+            if (ConstantHolder.INPUT.contains(event.getType()))  {
+                checkExpressionSyntax();}
+        });
+    }
```

Comment:
Sorry I lost here. Can't we just add new method with `@EventHandler` annotation mentioning element name/id to that we register handler as we do for most of other elements?

Reply:
@jomarko Unfortunately, GWT doesn't handle native `input` event. This is a way to register it and handle it.